### PR TITLE
Fix: CollectM failing  scala 3 jdk 11

### LINKED ISSUE #613: Type mismatch: java.lang.Object is not a subtype of io.netty.util.concurrent.Future
ZIO: 1.0.12; zhttp 1.0.0.0-RC17 works fine; with 1.0.0.0-RC18 I get:

```
Fiber failed.
An unchecked error was produced.
java.lang.BootstrapMethodError: bootstrap method initialization exception
	at java.base/java.lang.invoke.BootstrapMethodInvoker.invoke(BootstrapMethodInvoker.java:194)
	at java.base/java.lang.invoke.CallSite.makeSite(CallSite.java:307)
	at java.base/java.lang.invoke.MethodHandleNatives.linkCallSiteImpl(MethodHandleNatives.java:258)
	at java.base/java.lang.invoke.MethodHandleNatives.linkCallSite(MethodHandleNatives.java:248)
	at zhttp.service.HttpRuntime.unsafeRun$$anonfun$4$$anonfun$2$$anonfun$2(HttpRuntime.scala:24)
	at zio.ZIO$.effectTotal$$anonfun$1(ZIO.scala:2819)
	at zio.internal.FiberContext.evaluateNow(FiberContext.scala:344)
	at zio.internal.FiberContext.evaluateLater$$anonfun$1(FiberContext.scala:776)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: java.lang.invoke.LambdaConversionException: Type mismatch for instantiated parameter 0: class java.lang.Object is not a subtype of interface io.netty.util.concurrent.Future
	at java.base/java.lang.invoke.AbstractValidatingLambdaMetafactory.checkDescriptor(AbstractValidatingLambdaMetafactory.java:308)
	at java.base/java.lang.invoke.AbstractValidatingLambdaMetafactory.validateMetafactoryArgs(AbstractValidatingLambdaMetafactory.java:294)
	at java.base/java.lang.invoke.LambdaMetafactory.altMetafactory(LambdaMetafactory.java:503)
	at java.base/java.lang.invoke.BootstrapMethodInvoker.invoke(BootstrapMethodInvoker.java:138)
	... 10 more
```

What am I doing wrong?
Thanks!

### LINKED ISSUE #655: Http.collectM crashes on scala 3
A simple app like this:

```scala
import zio.*
import zhttp.http.*
import zhttp.service.Server

object Example extends App:

  val app = Http.collectM[Request] { case Method.GET -> !! / "hola" => ZIO succeed Response.text("hola") }

  def run(args: List[String]) = Server.start(8090, app).exitCode
```

when calling that endpoint it responds successfully but crashes immediately with:

```
Fiber failed.
An unchecked error was produced.
java.lang.BootstrapMethodError: bootstrap method initialization exception
	at java.base/java.lang.invoke.BootstrapMethodInvoker.invoke(BootstrapMethodInvoker.java:188)
	at java.base/java.lang.invoke.CallSite.makeSite(CallSite.java:315)
	at java.base/java.lang.invoke.MethodHandleNatives.linkCallSiteImpl(MethodHandleNatives.java:281)
	at java.base/java.lang.invoke.MethodHandleNatives.linkCallSite(MethodHandleNatives.java:271)
	at zhttp.service.HttpRuntime.unsafeRun$$anonfun$4$$anonfun$2$$anonfun$2(HttpRuntime.scala:24)
	at zio.ZIO$.effectTotal$$anonfun$1(ZIO.scala:2819)
	at zio.internal.FiberContext.evaluateNow(FiberContext.scala:344)
	at zio.internal.FiberContext.evaluateLater$$anonfun$1(FiberContext.scala:776)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: java.lang.invoke.LambdaConversionException: Type mismatch for dynamic parameter 0: class java.lang.Object is not a subtype of interface io.netty.util.concurrent.Future
	at java.base/java.lang.invoke.AbstractValidatingLambdaMetafactory.checkDescriptor(AbstractValidatingLambdaMetafactory.java:325)
	at java.base/java.lang.invoke.AbstractValidatingLambdaMetafactory.validateMetafactoryArgs(AbstractValidatingLambdaMetafactory.java:311)
	at java.base/java.lang.invoke.LambdaMetafactory.altMetafactory(LambdaMetafactory.java:535)
	at java.base/java.lang.invoke.BootstrapMethodInvoker.invoke(BootstrapMethodInvoker.java:147)
	... 10 more

Fiber:Id(1639675798870,6) was supposed to continue to:
  a future continuation at zhttp.service.HttpRuntime.6$$anonfun(HttpRuntime.scala:23)
  a future continuation at zio.Runtime.unsafeRunWith$$anonfun$2(Runtime.scala:214)

Fiber:Id(1639675798870,6) execution trace:
  at zio.ZIO$.effectTotal$$anonfun$1(ZIO.scala:2819)
  at zhttp.service.HttpRuntime.7$$anonfun(HttpRuntime.scala:23)
  at zio.ZIO.$times$greater$$anonfun$1(ZIO.scala:95)

Fiber:Id(1639675798870,6) was spawned by: <empty trace>
Fiber failed.
An unchecked error was produced.
java.lang.BootstrapMethodError: bootstrap method initialization exception
	at zhttp.service.HttpRuntime.unsafeRun$$anonfun$4$$anonfun$2$$anonfun$2(HttpRuntime.scala:24)
	at zio.ZIO$.effectTotal$$anonfun$1(ZIO.scala:2819)
	at zio.internal.FiberContext.evaluateNow(FiberContext.scala:344)
	at zio.internal.FiberContext.evaluateLater$$anonfun$1(FiberContext.scala:776)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)

Fiber:Id(1639675837156,8) was supposed to continue to:
  a future continuation at zhttp.service.HttpRuntime.6$$anonfun(HttpRuntime.scala:23)
  a future continuation at zio.Runtime.unsafeRunWith$$anonfun$2(Runtime.scala:214)

Fiber:Id(1639675837156,8) execution trace:
  at zio.ZIO$.effectTotal$$anonfun$1(ZIO.scala:2819)
  at zhttp.service.HttpRuntime.7$$anonfun(HttpRuntime.scala:23)
  at zio.ZIO.$times$greater$$anonfun$1(ZIO.scala:95)

Fiber:Id(1639675837156,8) was spawned by: <empty trace>
Fiber failed.
An unchecked error was produced.
java.lang.BootstrapMethodError: bootstrap method initialization exception
	at zhttp.service.HttpRuntime.unsafeRun$$anonfun$4$$anonfun$2$$anonfun$2(HttpRuntime.scala:24)
	at zio.ZIO$.effectTotal$$anonfun$1(ZIO.scala:2819)
	at zio.internal.FiberContext.evaluateNow(FiberContext.scala:344)
	at zio.internal.FiberContext.evaluateLater$$anonfun$1(FiberContext.scala:776)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)

Fiber:Id(1639675841972,10) was supposed to continue to:
  a future continuation at zhttp.service.HttpRuntime.6$$anonfun(HttpRuntime.scala:23)
  a future continuation at zio.Runtime.unsafeRunWith$$anonfun$2(Runtime.scala:214)

Fiber:Id(1639675841972,10) execution trace:
  at zio.ZIO$.effectTotal$$anonfun$1(ZIO.scala:2819)
  at zhttp.service.HttpRuntime.7$$anonfun(HttpRuntime.scala:23)
  at zio.ZIO.$times$greater$$anonfun$1(ZIO.scala:95)

Fiber:Id(1639675841972,10) was spawned by: <empty trace>

```

**Desktop:**
 - OS: macOS 10.15.7

**Additional context**
`build.sbt` file:

```scala
ThisBuild / scalaVersion := "3.1.0"

libraryDependencies += "dev.zio" %% "zio"   % "1.0.12"
libraryDependencies += "io.d11"  %% "zhttp" % "1.0.0.0-RC18"

```

### PR DESCRIPTION
**Pull request by amitksingh1490**

### CODE REVIEW

**Comment by amitksingh1490** [comment]
## 🚀🚀🚀 Benchmark Results 
 **Concurrency: 256** 
 **Requests/sec: 434329.95**

**Comment by amitksingh1490** [comment]
## 🚀🚀🚀 Benchmark Results 
 **Concurrency: 256** 
 **Requests/sec: 428583.67**

**Comment by amitksingh1490** [comment]
## 🚀🚀🚀 Benchmark Results 
 **Concurrency: 256** 
 **Requests/sec: 437259.37**

**Comment by amitksingh1490** [comment]
## 🚀🚀🚀 Benchmark Results 
 **Concurrency: 256** 
 **Requests/sec: 446853.31**

**Comment by amitksingh1490** [comment]
## 🚀🚀🚀 Benchmark Results 
 **Concurrency: 256** 
 **Requests/sec: 440157.24**

**Review by girdharshubham — APPROVED** [review]
This looks good to me.

**Comment by amitksingh1490** [comment]
## 🚀🚀🚀 Benchmark Results 
 **Concurrency: 256** 
 **Requests/sec: 423729.93**

File: zio-http/src/main/scala/zhttp/service/HttpRuntime.scala [hunk 1]
```diff
@@ -20,8 +20,8 @@ final class HttpRuntime[+R](strategy: HttpRuntime.Strategy[R]) {
     rtm
       .unsafeRunAsync(for {
         fiber <- program.fork
-        _     <- ZIO.effectTotal {
-          ctx.channel().closeFuture.addListener((_: AnyRef) => rtm.unsafeRunAsync_(fiber.interrupt): Unit)
+        _     <- ZIO.effect {
```
Comment (tusharmath) [inline zio-http/src/main/scala/zhttp/service/HttpRuntime.scala:23]: why this change?

File: zio-http/src/test/scala/zhttp/service/ServerSpec.scala [hunk 2]
```diff
@@ -152,6 +152,12 @@ object ServerSpec extends HttpRunnableSpec(8088) {
       testM("200 response with encoded path") {
         val actual = status(!! / "get%2Fsuccess")
         assertM(actual)(equalTo(Status.OK))
+      } +
+      testM("Multiple 200 response") {
+        val data = Gen.listOfN(5)(Gen.fromEffect(status(!! / "success").orElse(UIO(Status.INTERNAL_SERVER_ERROR))))
```
Comment (tusharmath) [inline zio-http/src/test/scala/zhttp/service/ServerSpec.scala:157]: - I don't think we need to use a `Gen` here and use the `repeat` operator on ZIO.
- We should consider larger numbers (eg 1024) for such use-cases. Essentially something that is larger than number of `threads` or `ports` etc.

File: zio-http/src/main/scala/zhttp/service/HttpRuntime.scala (hunk 1 above)
Comment (amitksingh1490) [reply zio-http/src/main/scala/zhttp/service/HttpRuntime.scala:23]: EffectTotal should be used when we know there won't be any exceptions, but the statement on line 24 can throw.

**Pull request merged**